src/protocol/vm/utils.rs

```diff
@@ -248,9 +255,9 @@ pub async fn get_code_for_contract(
     let connection_string = match connection_string {
         Some(url) => url,
         None => {
-            return Err(SimulationError::from(RpcError::InvalidRequest(
+            return Err(SimulationError::RetryDifferentInput(
```

Comment:
I think this should be Fatal no?

Reply:
I disagree here in particular - does the user not have the option to reset their `RPC_URL` and try again?

Reply:
Do you expect them to do that while the process is running?

Reply:
Not sure what you mean by this, are we expecting them to change the sell amount while the process is running if the sell amount is too high? I don't really see how this would be different. This doesn't by any means indicate a bug in our own code or the protocol code - so I feel calling it Fatal would be inaccurate

Reply:
I think this should be Fatal. An RPC url is a set-up issue, not a runtime issue... it shouldn't ever change while the program is running and needs external input by a human in order to fix it if it is wrong.
As for sell amount errors - yes, that's a runtime error and can easily be handled automatically by a program that recognises the error and reduces the sell amount to try again. This doesn't need human input to correct.